src/nvim/shada.c

```diff
@@ -2216,8 +2222,22 @@ static inline ShaDaWriteResult shada_read_when_writing(ShaDaReadDef *const sd_re
               }
               shada_free_shada_entry(&wms_entry->data);
             }
+          } else {
+            FOR_ALL_BUFFERS(buf) {
```

Comment:
Is there a more concise way to do this (look up buffer handle by buffer filename)? Not sure if this will be brittle in some unforeseen way.

Reply:
There is a `find_buffer` static function in shada.c, but it does caching so I don't know if it's necessary. Looks like the only thing it's doing differently is using `FNAMECMP` instead of `strequal`.